File: src/IsbInterop/Accessory/Wrappers/Contents.vb
Imports IsbInterop.Base

Namespace Accessory.Wrappers
    ''' <summary>
    ''' Обертка над IContents.
    ''' </summary>
    Friend Class Contents(Of TI As IObjectInfo)
        Inherits ForEach(Of TI)
        Implements IContents(Of TI)
        ''' <summary>
        ''' Конструктор.
        ''' </summary>
        ''' <param name="contentsObject">Объект IContents.</param>
        Public Sub New(contentsObject As Object)
            MyBase.New(contentsObject)
        End Sub
    End Class
End Namespace

File: src/IsbInterop/Collections/Proxies/Contents.cs
using IsbInterop.Base;

namespace IsbInterop.Collections.Proxies
{
  /// <summary>
  /// Обертка над IContents.
  /// </summary>
  internal class Contents<TI> : ForEach<TI>, IContents<TI> where TI : IObjectInfo
  {
    /// <summary>
    /// Конструктор.
    /// </summary>
    /// <param name="contentsObject">Объект IContents.</param>
    /// <param name="scope">Область видимости.</param>
    public Contents(object contentsObject, IScope scope) : base(contentsObject, scope) { }
  }
}
